src/parser/mod.rs

```diff
@@ -6830,7 +6830,15 @@ impl<'a> Parser<'a> {
                 let columns = self.parse_parenthesized_column_list(Mandatory, false)?;
                 self.expect_keyword(Keyword::REFERENCES)?;
                 let foreign_table = self.parse_object_name(false)?;
-                let referred_columns = self.parse_parenthesized_column_list(Mandatory, false)?;
+                // PostgreSQL allows foreign key columns to be optional
+                // (https://www.postgresql.org/docs/current/sql-createtable.html#SQL-CREATETABLE-PARMS-REFERENCES)
+                let parenthesized_column_list_optional = if dialect_of!(self is PostgreSqlDialect) {
```

Comment:
Since this behavior is a superset of the previous I think we can do without the dialect check and always set to `Optional`?

Also could we add a test case to [sqlparse_common.rs](https://github.com/apache/datafusion-sqlparser-rs/blob/main/tests/sqlparser_common.rs) demonstrating the new behavior?

Reply:
Good point. Made the change.